data_safe_haven/external/api/graph_api.py

```diff
@@ -1045,23 +1045,28 @@ def verify_custom_domain(
                         for nameserver in expected_nameservers
                     ):
                         self.logger.info(
-                            f"Verified that [green]{domain_name}[/] is registered as a custom Entra ID domain."
+                            f"[green]{domain_name}[/] DNS has been delegated correctly."
                         )
                         break
                 self.logger.warning(
-                    f"Domain [green]{domain_name}[/] is not currently registered as a custom Entra ID domain."
+                    f"[green]{domain_name}[/] DNS is not delegated correctly."
                 )
                 # Prompt user to set domain delegation manually
-                docs_link = "https://learn.microsoft.com/en-us/azure/dns/dns-delegate-domain-azure-dns#delegate-the-domain"
                 self.logger.info(
-                    f"To proceed you will need to delegate [green]{domain_name}[/] to Azure ({docs_link})"
+                    f"To proceed you will need to delegate [green]{domain_name}[/] to specific Azure nameservers"
                 )
                 ns_list = ", ".join([f"[green]{n}[/]" for n in expected_nameservers])
                 self.logger.info(
-                    f"You will need to create NS records pointing to: {ns_list}"
+                    f"Create {len(ns_list)} [green]NS[/] records for [green]{domain_name}[/] with values: {ns_list}"
```

Comment:
Should we be even more clear here?

`Create an NS record for {subdomain} in {parent_domain} with these nameservers: {ns_list}`

We'd have to construct `subdomain` and `parent_domain` but I think this will (always?) be `domain_name.split(".")[0]` and `"".join(domain_name.split(".")[1:])`.

Reply:
_Maybe_.

I did think about that, but I'm not sure if it assumed more about our users DNS set up than we can.

I think what we actually require is that all requests to `sre.tre.com` resolve to the Azure nameservers. Which isn't the same as saying that the nameservers for `tre.com` hold those NS records.

Reply:
Sure, but the registrar for `tre.com` has to have an NS record for `sre` which is the place we need users to add these records. I think we can safely decompose `sre.tre.com` into `sre` (subdomain) and `tre.com` (parent domain) right?

Reply:
What I'm saying is, `tre.com` doesn't need to have NS records for `sre.tre.com` _to_ Azure nameservers. It could point to some other nameservers, as long as those all eventually resolve to the Azure nameservers.

I'm not really sure _why_ you would want to do that, but it isn't really my area of expertise.

I suppose what I'm trying to figure out is,

Is the benefit of being a bit more specific ("create this record, and create it here" rather than "create this record, and I assume you know where to do that") worth the cost of saying something that might not be quite true 🤔.

Maybe this is one of those cases where if you have some more complex DNS setup, you probably know what you are doing anyway?

Reply:
Yeah, maybe we could have something like "you'll need to ensure {subdomain} is delegated to the nameservers: {ns_list}. If you're not sure what this means then do X"

Reply:
Hmm, I'm less keen on that because I don't feel confident I know what they _should_ do.

Maybe,

`Create a NS records for {subdomain} (for example in the {parent_domain} zone) ...`